chisel/src/driver.rs

```diff
@@ -184,6 +195,235 @@ impl ChiselDriver {
+                        return Err(DriverError::InvalidField(name, "preset".to_string()));
+                    }
+                } else {
+                    chisel_debug!(1, "verifyexports missing field 'preset'");
+                    return Err(DriverError::MissingRequiredField(
+                        name,
+                        "preset".to_string(),
+                    ));
+                }
+            }
+            "verifyimports" => {
+                if let Some(preset) = module.options().get("preset") {
+                    let verifyimports = VerifyImports::with_preset(preset.as_str());
+                    if let Ok(verifyimports) = verifyimports {
+                        let module_result = verifyimports.validate(&wasm);
+                        ModuleResult::Validator(name, module_result)
+                    } else {
+                        chisel_debug!(1, "verifyimports given invalid preset");
+                        return Err(DriverError::InvalidField(name, "preset".to_string()));
+                    }
+                } else {
+                    chisel_debug!(1, "verifyimports missing field 'preset'");
+                    return Err(DriverError::MissingRequiredField(
+                        name,
+                        "preset".to_string(),
+                    ));
+                }
+            }
+            #[cfg(feature = "binaryen")]
+            "binaryenopt" => {
+                if let Some(preset) = module.options().get("preset") {
+                    let binaryenopt = BinaryenOptimiser::with_preset(preset.as_str());
+                    if let Ok(binaryenopt) = binaryenopt {
+                        match binaryenopt.translate(wasm) {
+                            Ok(new_wasm) => {
+                                let did_mutate = if let Some(new_wasm) = new_wasm {
+                                    *wasm = new_wasm;
+                                    true
+                                } else {
+                                    false
+                                };
+
+                                ModuleResult::Translator(name, Ok(did_mutate))
+                            }
+                            Err(e) => ModuleResult::Translator(name, Err(e)),
+                        }
+                    } else {
+                        chisel_debug!(1, "binaryenopt given invalid preset");
+                        return Err(DriverError::InvalidField(name, "preset".to_string()));
+                    }
+                } else {
+                    chisel_debug!(1, "binaryenopt missing field 'preset'");
+                    return Err(DriverError::MissingRequiredField(
+                        name,
+                        "preset".to_string(),
+                    ));
+                }
+            }
+            other => {
+                return Err(DriverError::ModuleNotFound(name.clone(), other.to_string()));
```

Comment:
No need for clone here?

Reply:
It is needed, since it's in a scope where name is borrowed. See line 206

Reply:
NVM, it is actually needed because we use `other.to_string()` as the second argument.